src/omv/sensors/paj6100.c

```diff
@@ -0,0 +1,788 @@
+/*
+ * File: paj6100.c
+ * Created Date: Tuesday, May 25th 2021, 10:45:35 am
+ * Author: Lake Fu
+ * -----
+ * Last Modified: Tuesday June 15th 2021 2:54:43 pm
+ * Modified By: Lake Fu at <lake_fu@pixart.com>
+ * -----
+ * MIT License
+ *
+ * Copyright (c) 2021 Pixart Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ * -----
+ * HISTORY:
+ * Date      	By	Comments
+ * ----------	---	----------------------------------------------------------
+ */
+#include "omv_boardconfig.h"
+#include "paj6100.h"
+#if (OMV_ENABLE_PAJ6100 == 1)
```

Comment:
```suggestion
#if (OMV_ENABLE_PAJ6100 == 1)
#include "paj6100.h"
```

Reply:
That will cause a build fail, I need stdbool.h, and it put in header.

Reply:
@iabdalkader 
Check this, [https://travis-ci.com/github/openmv/openmv/jobs/513966439#L991](url)

Reply:
Link is not working but I see what you mean, but if you use the `OMV_ENABLE_PAJ6100` macro in sensor.c you don't need to define these functions (and need stdbool) if the sensor is disabled.

Reply:
I know what you mean, but I don't really like to make a function become undefined reference, I don't like a link time error.. XD.
Yes I understand that header should not be used when OMV_ENABLE_PAJ6100 disabled, but it actually be included by sensor.c.

And I think that will be ok even it is included because the compiler optimization may remove it.

But, it's your call. Just let me know what you final decision.

Reply:
Hi, there won't be any compiler errors because you don't need to define these functions at all if the sensor is disabled:
```
#else
bool paj6100_detect(sensor_t *sensor) { return false; }
int paj6100_init(sensor_t *sensor) { return 0; }
```
If the sensor is disabled paj6100_detect and paj6100_init are Not called from sensor.c

Reply:
No, it is not about compile time error.
If we do so, the header still declare those 2 functions, but those are undefined the references.
That means, anybody can call those 2 functions without "compile time" error, but throw error when link time. 

> If the sensor is disabled paj6100_detect and paj6100_init are Not called from sensor.c

Yes, it is.
But I don't like to declare functions in header (.h) without the body (so-called "implementation" etc...).
It just a coding style to avoid I do something bad.. XD.

In my mind, if I remove those empty functions, I will add "#ifdef OMV_ENABLE_PAJ6100" to header (paj6100.h) to skip the declare also. (My English is not very good, I hope that you will understand what I mean.)
```
#else
bool paj6100_detect(sensor_t *sensor) { return false; }
int paj6100_init(sensor_t *sensor) { return 0; }
```

```
#ifndef __PAJ6100_H__
#define __PAJ6100_H__
#ifdef OMV_ENABLE_PAJ6100  <---- THIS ~     
#include <stdbool.h>;
#include "sensor.h"
#define PAJ6100_XCLK_FREQ 6000000
bool paj6100_detect(sensor_t *sensor);
int paj6100_init(sensor_t *sensor);
#endif
#endif
```
Can I do that?

Reply:
> But I don't like to declare functions in header (.h) without the body (so-called "implementation" etc...).
It just a coding style to avoid I do something bad.. XD.

That's okay will accept that, these functions are not called when the sensor is disabled there won't be any link-time error either. 

Please remove this:
```
#else
bool paj6100_detect(sensor_t *sensor) { return false; }
int paj6100_init(sensor_t *sensor) { return 0; }
``
And keep the functions declared in the header that's fine.

And now you can place the header include after the macro:
```
#if (OMV_ENABLE_PAJ6100 == 1)
#include "paj6100.h"
```

Note it's best to finish up this PR soon because I need to do a new release and include this in the new release, so please resolve all the remaining issues as soon as possible.

Reply:
Ok, I will finish it.